### PR TITLE
ff: Filter duplicate registrations in subtheory.

### PR DESCRIPTION
**Pull request by alex-ozdemir**

### CODE REVIEW

**Review by ajreynol — CHANGES_REQUESTED** [review]
Thanks, this makes sense now.  The proper fix is actually to make `d_leaves` *SAT* context dependent instead of user context dependent. 
Preregistration already ensures terms are registered at most once in a SAT context, so a set is not necessary.  

Thus the fix should just be one line change.

File: src/theory/ff/sub_theory.cpp [hunk 1]
```diff
@@ -39,6 +40,7 @@ SubTheory::SubTheory(Env& env, FfStatistics* stats, Integer modulus)
       context::ContextNotifyObj(context()),
       d_facts(context()),
       d_leaves(userContext()),
```
Comment (ajreynol) [inline src/theory/ff/sub_theory.cpp:42]:
```suggestion
      d_leaves(context()),
```

**Comment by ajreynol** [comment]
This isn't a blocker for me at the moment, how about we revisit this when its convenient for you.  For now, we can leave the PR open.

**Comment by alex-ozdemir** [comment]
I'm closing this in favor of #9486.

**Pull request closed without merging**